core/lib/types/src/api/mod.rs

```diff
@@ -714,6 +714,8 @@ pub struct BlockDetails {
 #[serde(rename_all = "camelCase")]
 pub struct L1BatchDetails {
     pub number: L1BatchNumber,
+    pub commitment: Option<H256>,
+    pub rollup_last_leaf_index: Option<u64>,
```

Comment:
IMO, we shouldn't add these fields to this endpoint only because of genesis changes. Maybe it's possible to get this data from L1 contract for genesis batch? If not maybe it's worth adding a separate endpoint

Reply:
Tbh, i was thinking that these information is useful and we can keep it in this method